### PR TITLE
[T4A4][W09-A4]Muhammad Mustaqiim Bin Muhar

### PR DESCRIPTION
**Pull request by mustaqiimuhar**

### CODE REVIEW

**Comment by mustaqiimuhar** [comment]
Ready for review.

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -56,4 +56,10 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+	@Override
+	public String getPrintableString(Printable... printables) {
+		// TODO Auto-generated method stub
+		return "Address: " + toString();
+	}
```
Comment (hwkchia) [inline src/seedu/addressbook/data/person/Address.java:64]: Check `isPrivate`? Likewise for email and phone.

File: src/seedu/addressbook/data/person/Printable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package seedu.addressbook.data.person;
+
+public interface Printable {
```
Comment (hwkchia) [inline src/seedu/addressbook/data/person/Printable.java:3]: Header comment?

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 3]
```diff
@@ -83,4 +85,13 @@ default String getAsTextHidePrivate() {
...
     }
+    
+    default String getPrintableString(Printable...printables){
+    	ArrayList<String> allPrintables = new ArrayList<String>();
+    	for (Printable printable : printables) {
+    		allPrintables.add(printable.getPrintableString());
+    	}
+    	String joinPrintables = String.join(", ", allPrintables);
+    	return joinPrintables;
+    }
```
Comment (hwkchia) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:96]: 👍 for using the join method

**Review by hwkchia — COMMENTED** [review]
Good job on applying ISP

**Comment by hwkchia** [comment]
Ack by closing PR

**Pull request closed without merging**